src/ui/localization/translator.ts

```diff
@@ -0,0 +1,246 @@
+import en from "./locale/en.json";
+import es from "./locale/es.json";
+import it from "./locale/it.json";
+import type { PeriodUnit } from "../../helpers/duration-helper";
+
+/**
+ * Custom translations to be used in the purchase flow.
+ * This class allows you to override the default translations used in the purchase flow.
+ * The main level keys are the locale codes and the values are objects with the same keys as the default translations.
+ * @public
+ * @example
+ * This example will override the default translation for the email step title in the English locale.
+ * ```typescript
+ * const customTranslations = {
+ *  en: {
+ *    "state_needs_auth_info.email_step_title": "Billing email",
+ *   }
+ * }
+ * ```
+ */
+export type CustomTranslations = Record<string, Record<string, string>>;
```

Comment:
I think this would also benefit from having some sort of enum, for the supported languages and localization keys.